src/module/macros/overcharge.ts

```diff
@@ -0,0 +1,66 @@
+// Import TypeScript modules
+import { LANCER } from "../config";
+import { getAutomationOptions } from "../settings";
+import type { LancerActor } from "../actor/lancer-actor";
+import type { LancerOverchargeMacroData } from "../interfaces";
+import { getMacroSpeaker } from "./util"
+import { renderMacroTemplate } from "./render"
+
+const lp = LANCER.log_prefix;
+
+export async function prepareOverchargeMacro(a: string) {
+  // Determine which Actor to speak as
+  let actor = getMacroSpeaker(a);
+  if (!actor) return;
+
+  // Validate that we're overcharging a mech
+  if (!actor.is_mech()) {
+    ui.notifications!.warn(`Only mechs can overcharge!`);
+    return;
+  }
+
+  // And here too... we should probably revisit our type definitions...
+  let rollText = actor.getOverchargeRoll();
+  if (!rollText) {
+    ui.notifications!.warn(`Error in getting overcharge roll...`);
+    return;
+  }
+
+  // Prep data
+  let roll = await new Roll(rollText).evaluate({ async: true });
+
+  let mech = actor.data.data.derived.mm!;
+
+  let mData: LancerOverchargeMacroData = {
+    level: mech.OverchargeCount,
+    roll: roll,
+  };
+
+  // Assume we can always increment overcharge here...
+  mech.OverchargeCount = Math.min(mech.OverchargeCount + 1, 3);
+
+  // Only increase heat if we haven't disabled it
+  if (getAutomationOptions().overcharge_heat) {
+    mech.CurrentHeat = mech.CurrentHeat + roll.total!;
+  }
+
+  await mech.writeback();
+
+  return rollOverchargeMacro(actor, mData);
+}
+
+async function rollOverchargeMacro(actor: LancerActor, data: LancerOverchargeMacroData) {
```

Comment:
Naming and functionality is counterintuitive here. `prepareOverchargeMacro` does the roll, `rollOverchargeMacro` renders it.